kepler-core/src/keys.rs

```diff
@@ -0,0 +1,74 @@
+use kepler_lib::{
+    libipld::cid::multihash::{Blake3_256, Hasher},
+    resource::OrbitId,
+};
+use libp2p::{
+    identity::{
+        ed25519::{Keypair as EdKP, SecretKey},
+        DecodingError, Keypair, PublicKey,
+    },
+    PeerId,
+};
+use sea_orm_migration::async_trait::async_trait;
+use std::error::Error as StdError;
+
+#[async_trait]
+pub trait Secrets {
+    type Error: StdError;
+    async fn get_keypair(&self, orbit: &OrbitId) -> Result<Keypair, Self::Error>;
+    async fn get_pubkey(&self, orbit: &OrbitId) -> Result<PublicKey, Self::Error> {
+        Ok(self.get_keypair(orbit).await?.public())
+    }
+    async fn stage_keypair(&self, orbit: &OrbitId) -> Result<PublicKey, Self::Error>;
+    async fn save_keypair(&self, orbit: &OrbitId) -> Result<(), Self::Error>;
```

Comment:
What's the purpose of this method if the goal is to always re-generate it?

Reply:
I wanted to account for an alternative impl which would generate them randomly and store them temporarily until the orbit is actually created (as before)